flag.go

```diff
@@ -437,7 +437,17 @@ func (f *Flag) defaultIsZeroValue() bool {
 	case *intSliceValue, *stringSliceValue, *stringArrayValue:
 		return f.DefValue == "[]"
 	default:
-		return true
+		switch f.Value.String() {
```

Comment:
Why does this use `f.Value.String()` instead of `f.DefValue`?

In cases where a caller supplies a value for a custom flag, and that value is not a zero value, this expression yields the wrong result: the _default value_ of the flag is a zero value, but the _set value_ of the flag is not, so the help winds up printing something like
> (default )

per [this line](https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/flag.go#L735) when it should have omitted mention of a default value.